File: complementing_a_strand_of_DNA/complementing_a_strand_of_DNA.R
reverse_compliment <- function(target_sequence){
  
  complement <- c('A'='T', 'C'='G', 'G'='C', 'T'='A')
  
  split_sequence <- unlist(strsplit(target_sequence, split = ""))
  reversed_sequence <- rev(split_sequence)
  
  complimented_sequence <- sapply(reversed_sequence, FUN = function(x) { complement[x] })
  complimented_sequence <- paste(complimented_sequence, collapse = "")
  
  return(complimented_sequence)
}

sequence <- readLines("rosalind_revc.txt")
sequence <- unlist(sequence)

reversed <- reverse_compliment(sequence)
cat(reversed)

File: complementing_a_strand_of_DNA/complementing_a_strand_of_DNA.py
def main():
    with open("rosalind_revc.txt") as file:
        sequence = file.read().strip()

    reversed_sequence = reverse_complement(sequence)
    print(reversed_sequence)


def reverse_complement(seq):
    complement = {'A': 'T', 'C': 'G', 'G': 'C', 'T': 'A'}
    return "".join(complement[base] for base in reversed(seq))


if __name__ == "__main__":
    main()
